### PR TITLE
update quantization tutorial by introudcing x86 backend

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Because the latest Pytorch introduces  [x86 backend](https://github.com/pytorch/pytorch/issues/83888) to unify the x86 CPUs quantization, this PR updates the quantization tutorial by setting `x86` backend as default backend on x86 cpus.

cc @jerryzh168 @z-a-f @vkuzo

### CODE REVIEW

File: beginner_source/vt_tutorial.py [hunk 1]
```diff
@@ -138,8 +138,8 @@
 # Now run the code below:
 #
 
-# Use 'fbgemm' for server inference and 'qnnpack' for mobile inference
-backend = "fbgemm" # replaced with qnnpack causing much worse inference speed for quantized model on this notebook
+# Use 'x86' for server inference (th old 'fbgemm' is still available but 'x86' is the recommended default) and 'qnnpack' for mobile inference.
```
Comment (jgong5) [inline beginner_source/vt_tutorial.py:141]:
```suggestion
# Use 'x86' for server inference (the old 'fbgemm' is still available but 'x86' is the recommended default) and 'qnnpack' for mobile inference.
```

**Review by jgong5 — CHANGES_REQUESTED** [review]

**Comment by vkuzo** [comment]
cc @jerryzh168 , do we know the status of the x86 issues with respect to PyTorch v1.13?  Did we cherry-pick backing out the x86 as default, and do we have a timeline on fix to x86 backend?

**Comment by jerryzh168** [comment]
> cc @jerryzh168 , do we know the status of the x86 issues with respect to PyTorch v1.13? Did we cherry-pick backing out the x86 as default, and do we have a timeline on fix to x86 backend?

in PyTorch 1.13 we have reverted back to using fbgemm as default backend. cherry-pick is merged: https://github.com/pytorch/pytorch/issues/86312#issuecomment-1275329963

I'm trying out the fix now: https://github.com/pytorch/pytorch/pull/86876

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
let's wait until the problem is resolved

**Review by jerryzh168 — APPROVED** [review]
thanks, looks like there are some broken ci tests, please make sure everything is green before landing

**Comment by XiaobingSuper** [comment]
@jerryzh168, please help merge it,

**Comment by jerryzh168** [comment]
@svekars can you help merging this? I'm not sure how to merge tutorial PRs

**Comment by svekars** [comment]
@XiaobingSuper Hello, can you please provide some context around this change? As far as I understand this change is for the PyTorch release 2.0. We typically merge such PRs on the day of the release because we don't version tutorials. I'm not seeing any issues at the moment and I will also run this PR against the RC branch when it becomes available. No actions needed on your side as of now. Does this sound good?

**Comment by XiaobingSuper** [comment]
@svekars , yes, this is for PyTorch 2.0, we introduced this feature for [Unified quantization backend for x86 CPU platforms](https://github.com/pytorch/pytorch/issues/83888). it is ok to merge this PR on the 2.0 release data, thanks!

**Comment by XiaobingSuper** [comment]
@svekars , is it ready to merge this PR for PT 2.0 release? Thanks!

**Review by msaroufim — APPROVED** [review]
Looks good but might be worth explaining in a few sentences to users why the new x86 backend is better

File: recipes_source/quantization.rst [hunk 2]
```diff
@@ -98,7 +98,7 @@ After this, running `print_model_size(model_static_quantized)` shows the static
 A complete model definition and static quantization example is `here <https://pytorch.org/docs/stable/quantization.html#quantization-api-summary>`_. A dedicated static quantization tutorial is `here <https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html>`_.
 
 .. note::
-  To make the model run on mobile devices which normally have arm architecture, you need to use `qnnpack` for `backend`; to run the model on computer with x86 architecture, use `fbgemm`.
+  To make the model run on mobile devices which normally have arm architecture, you need to use `qnnpack` for `backend`; to run the model on computer with x86 architecture, use `x86`` (the old `fbgemm` is still available but 'x86' is the recommended default).
```
Comment (jgong5) [inline recipes_source/quantization.rst:101]:
```suggestion
  To make the model run on mobile devices which normally have arm architecture, you need to use `qnnpack` for `backend`; to run the model on computer with x86 architecture, use `x86`` (the old `fbgemm` is still available but 'x86' is the recommended default which leverages the goodness of both oneDNN and fbgemm libraries and runs faster).
```

**Review by jgong5 — COMMENTED** [review]
@msaroufim Thanks for the review and suggestion. I added a suggest edit to one of the changes to elaborate on the benefit. Does that sound good to you? If so, is it fine to add this to all the changes?

**Pull request merged**